ckanext/dalrrd_emc_dcpr/helpers.py

```diff
@@ -342,3 +342,19 @@ def get_dcpr_requests_approved_by_nsif(request_origin):
         "dcpr_request_list_awaiting_csi_moderation"
     )({"request_origin": request_origin})
     return dcpr_requests_approved_by_nsif
+
+
+def is_dcpr_request(package):
+
+    for extra in package.get("extras") or []:
+        if extra.get("key") == "origin" and extra.get("value") == "DCPR":
```

Comment:
will this drop a keyError if extras is not in the package ?

Reply:
@Mohab25 the `get` returns None if the key doesn't exist. It doesn't throw any error.